src/Common/tests/System/RealFormatterTestsBase.netcoreapp.cs

```diff
@@ -14,29 +14,118 @@ public abstract class RealFormatterTestsBase
         protected abstract string InvariantToStringDouble(double d, string format);
         protected abstract string InvariantToStringSingle(float f, string format);
 
+        [Theory]
+        [InlineData("#,###.00", double.Epsilon, ".00")]
+        [InlineData("#,###.00", double.MaxValue, "179,769,313,486,232,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000.00")]
+        [InlineData("#,###.00", Math.E, "2.72")]
+        [InlineData("#,###.00", Math.PI, "3.14")]
+        [InlineData("#,###.00", 0.0, ".00")]
+        [InlineData("#,###.00", 0.0046, ".00")]
+        [InlineData("#,###.00", 0.005, ".01")]
+        [InlineData("#,###.00", 0.125, ".12")]
+        [InlineData("#,###.00", 0.5, ".50")]
+        [InlineData("#,###.00", 0.51, ".51")]
+        [InlineData("#,###.00", 0.56789, ".57")]
+        [InlineData("#,###.00", 0.84551240822557006, ".85")]
+        [InlineData("#,###.00", 1.0, "1.00")]
+        [InlineData("#,###.00", 505.0, "505.00")]
+        [InlineData("#,###.00", 506.0, "506.00")]
+        [InlineData("#,###.00", 545.0, "545.00")]
+        [InlineData("#,###.00", 545.1, "545.10")]
+        [InlineData("#,###.00", 555.0, "555.00")]
+        [InlineData("#,###.00", 65747.125, "65,747.12")]
+        [InlineData("#,###.00", 1844674407370955.25, "1,844,674,407,370,960.00")]
+        public void TestFormatterDouble_Custom(string format, double value, string expectedResult) => TestFormatterDouble_Standard(value, format, expectedResult);
+
         [Theory]
         [InlineData(double.Epsilon, "¤0.00")]
         [InlineData(double.MaxValue, "¤179,769,313,486,231,570,814,527,423,731,704,356,798,070,567,525,844,996,598,917,476,803,157,260,780,028,538,760,589,558,632,766,878,171,540,458,953,514,382,464,234,321,326,889,464,182,768,467,546,703,537,516,986,049,910,576,551,282,076,245,490,090,389,328,944,075,868,508,455,133,942,304,583,236,903,222,948,165,808,559,332,123,348,274,797,826,204,144,723,168,738,177,180,919,299,881,250,404,026,184,124,858,368.00")]
         [InlineData(Math.E, "¤2.72")]
         [InlineData(Math.PI, "¤3.14")]
         [InlineData(0.0, "¤0.00")]
         [InlineData(0.0046, "¤0.00")]
+        [InlineData(0.005, "¤0.01")]
```

Comment:
Tests covering `0.005`, `0.5`, `0.51`, `0.56789`, `505`, `506`, `545`, `545.1`, `555`, and `65747.125` were added to each format test.

These test values that either have "round to even" semantics or appear to have "round to even" semantics but actually don't due to the nearest representable value being just under/just over (and so not actually being a "tie").

For example, `0.005` is actually `~0.0050000000000000001`, so it should always round up. But `0.5` is exactly `0.5` and so it should always round down.

For `single`, however. `0.005` is actually `~0.00499999988824129105`, so it should always round down. `0.5` is still exactly `0.5` and should also round down.